### PR TITLE
PCA Modifications

### PR DESCRIPTION
**Pull request by owerbat**
* n_samples < n_features case in PCA enabled
* PCA patching simplified

### CODE REVIEW

File: daal4py/sklearn/decomposition/_pca.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                 svd_solver='auto', tol=0.0, iterated_power='auto',
+                 random_state=None):
+        self.n_components = n_components
+        self.copy = copy
+        self.whiten = whiten
+        self.svd_solver = svd_solver
+        self.tol = tol
+        self.iterated_power = iterated_power
+        self.random_state = random_state
+
```
Comment (PetrovKP) [inline daal4py/sklearn/decomposition/_pca.py:57]: del line

**Comment by owerbat** [comment]
http://intel-ci.intel.com/eb349308-12aa-f1c8-9437-8cdcd4b723a1

File: daal4py/sklearn/decomposition/_pca.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                                 "was of type=%r"
+                                 % (n_components, type(n_components)))
+
+
+    def _fit_full_daal4py(self, X, n_components):
+        n_samples, n_features = X.shape
+        n_sf_min = min(n_samples, n_features)
+
+        if n_components == 'mle':
+            daal_n_components = n_features
```
Comment (PetrovKP) [inline daal4py/sklearn/decomposition/_pca.py:82]: What is the difference from the number of components from [here](https://github.com/IntelPython/daal4py/pull/383/files#diff-add3ec45396c9436de3ce96cd3c2d9fb5823b3e09c6a798b877cf4efdaf99370R108)?

File: daal4py/sklearn/decomposition/_pca.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    from sklearn.decomposition.pca import PCA as PCA_original
+
+if sklearn_check_version('0.23'):
+    from sklearn.decomposition._pca import _infer_dimension
+elif sklearn_check_version('0.22'):
+    from sklearn.decomposition._pca import _infer_dimension_
+else:
+    from sklearn.decomposition.pca import _infer_dimension_
+
+
```
Comment (PetrovKP) [inline daal4py/sklearn/decomposition/_pca.py:45]: Why didn't you transfer the `_fit` method? It seems to me that you can also edit there + the information about the conformance will not be distorted.

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/eb35c224-8d27-f17a-96c2-8cdcd4b723a1?

**Review by PetrovKP — APPROVED** [review]

**Pull request merged**